cmd/timestamp-server/app/root.go

```diff
@@ -75,34 +73,9 @@ func init() {
 	rootCmd.PersistentFlags().String("file-signer-key-path", "", "Path to file containing PEM-encoded private key. Supported formats include PKCS#1, PKCS#8, and RFC5915 for EC")
 	rootCmd.PersistentFlags().String("file-signer-passwd", "", "Password to decrypt private key")
 
-	rootCmd.PersistentFlags().Uint16("port", 3000, "Port to bind to")
-
 	if err := viper.BindPFlags(rootCmd.PersistentFlags()); err != nil {
 		log.Logger.Fatal(err)
 	}
-
-	log.Logger.Debugf("pprof enabled: %v", enablePprof)
-	// Enable pprof
-	if enablePprof {
```

Comment:
After some locally running the server with `--enable-pprof=true`, I don't think this block will ever be entered. The default `false` value of `enablePprof` is always used because the value of the flag is never evaluated via viper with `viper.GetBool("enable-pprof")`.

To solve this, I created a new `server` package that contains a pprof server constructor. This constructor is called in `timestamp-server/app/serve.go` after the flag's value is evaluated in that file.

I would like to create a constructor for the restapi server in this new package and call that in `timestamp-server/app/serve.go` as well but that can be done in a follow up PR if preferred to keep this one smaller.

Reply:
Good catch! interested as to why the value is never evaluated.